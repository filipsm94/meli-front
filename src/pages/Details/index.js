import React, { useEffect, useState } from 'react';
import CurrencyFormat from 'react-currency-format';
import { useParams } from 'react-router-dom';
import { searchProductById } from '../../services/searchProducts';
import Spinner from '../../components/Spinner';
import classes from './details.module.scss';

const DetailsPage = () => {
    const { id } = useParams();
    const [product, setProduct] = useState({})
    const [loading, setLoading] = useState(false);

    useEffect(()=>{
        setLoading(true);
        searchProductById(id).then(({item}) => {
            setProduct(item);
        }).catch().finally(() => setLoading(false));;
    },[id]);

    if (loading) return <Spinner /> ;

    return (
        <div className={classes.Card}>
            <div className={classes.Flex}>
                <div className={classes.Image}>
                    <img src={product.picture} alt={product.title} />
                    <div className={classes.Description}>
                        <h1>Descripcion del Producto</h1>
                        <p>{product.description}</p>
                        <ul className={classes.Columns}>
                            {
                                product.features ? product.features.map((feature, i)=> <li key={i}>{feature.name} : {feature.value_name}</li> ): null
                            }
                        </ul>
                    </div>
                </div>
                <div className={classes.Purchase}>
                    <p>{product.condition ? 'Nuevo': 'Usado'} - {product.sold_quantity ? product.sold_quantity: 0 } vendidos</p>
                    <p>{product.title}</p>
                    {product.price  ? <p>
                        <CurrencyFormat className={classes.Price} value={product.price.amount} displayType={'text'} thousandSeparator={true} prefix={'$ '} suffix={` ${product.price.currency}`} />
                    </p> : null}
                    <button className={classes.Btn}>Comprar</button>
                </div>
            </div>
        </div>
    );
}

export default DetailsPage