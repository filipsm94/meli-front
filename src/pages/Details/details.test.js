import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import Details from './index';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () =>({id:'MVAF65'})
}));

jest.mock('../../services/searchProducts', () => {
  const {responseProductDetailMock} = require('../../mocks/products.mock');
  return {
    searchProductById: (id) => Promise.resolve({ ...responseProductDetailMock }) // arreglar esto, este es el mock de productos y no de productodetalle
  }
});



describe('Details Component', () => {
  test('renders', () => {
    const { asFragment  } = render(<Details />);
    expect(asFragment()).toMatchSnapshot();
  });

})
