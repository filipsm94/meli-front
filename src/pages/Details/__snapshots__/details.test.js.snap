// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component renders 1`] = `
<DocumentFragment>
  <div
    class="Loader"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</DocumentFragment>
`;
